src/mixins/payment-mixin.ts

```diff
@@ -339,6 +339,20 @@ export class PaymentMixin extends Mixins(ActionMixin) {
       // await paymentModule.setPaymentReceipt(sbcPayment.receipts[0]) // FUTURE: verify that new code === old code
       await this.setPaymentRequest(data)
 
+      // delete previous incomplete payments after the new payment record created
+      const allPayments = await this.getNameRequestPayments(nrId, {})
+      if (allPayments) {
+        for (let i = 0; i < allPayments.length; i++) {
+          if (
+            allPayments[i].payment.id !== payment.id &&
+            allPayments[i].payment.payment_action === action &&
+            allPayments[i].sbcPayment.statusCode === SbcPaymentStatus.CREATED
```

Comment:
Are these caught gracefully in the `try/catch` in the event one of these values is not available?
For instance, if `allPayments[i].payment.payment_action` is missing `payment`. 

If not, optional chaining might a solution. Up to you!
eg `allPayments[i].payment?.payment_action`